src/components/CompareResults/loader.ts

```diff
@@ -115,34 +143,61 @@ async function fetchAllFakeCompareResults() {
 export async function loader({ request }: { request: Request }) {
   const url = new URL(request.url);
   const useFakeData = url.searchParams.has('fakedata');
-  const baseRev = url.searchParams.get('baseRev');
-  const baseRepo = url.searchParams.get('baseRepo') as
-    | Repository['name']
-    | null;
-  const newRevs = url.searchParams.getAll('newRev');
-  const newRepos = url.searchParams.getAll('newRepo') as Repository['name'][];
-  const framework = url.searchParams.get('framework');
 
-  let resultsForAllRevs;
   if (useFakeData) {
-    resultsForAllRevs = await fetchAllFakeCompareResults();
-  } else {
-    resultsForAllRevs = await checkValuesAndFetchCompareResultsOnTreeherder({
+    const results = await fetchAllFakeCompareResults();
+    // They're all based on the same rev
+    const baseRev = results[0][0].base_rev;
+    // And the same repository
+    const baseRepo = results[0][0].base_repository_name;
+
+    const newRevs = fakeCommitHashes;
+    const newRepos = results.map((result) => result[0].new_repository_name);
+    const frameworkId = results[0][0].framework_id;
+
+    return {
+      results,
       baseRev,
       baseRepo,
       newRevs,
       newRepos,
-      framework,
-    });
+      framework: frameworkId,
+    };
   }
 
+  const baseRevFromUrl = url.searchParams.get('baseRev');
+  const baseRepoFromUrl = url.searchParams.get('baseRepo') as
+    | Repository['name']
+    | null;
+  const newRevsFromUrl = url.searchParams.getAll('newRev');
+  const newReposFromUrl = url.searchParams.getAll(
+    'newRepo',
+  ) as Repository['name'][];
+  const frameworkFromUrl = url.searchParams.get('framework');
+
+  const { baseRev, baseRepo, newRevs, newRepos, frameworkId } = checkValues({
+    baseRev: baseRevFromUrl,
+    baseRepo: baseRepoFromUrl,
+    newRevs: newRevsFromUrl,
+    newRepos: newReposFromUrl,
+    framework: frameworkFromUrl,
+  });
+
+  const resultsForAllRevs = await fetchCompareResultsOnTreeherder({
```

Comment:
This is so much simpler and cleaner. Thanks so much for revamping this structure! Really liking this react-router API.